### PR TITLE
Require Swift 5.0

### PR DESCRIPTION
**Pull request by fabianfett**
Most Swift server libraries require at least Swift 5.0. Since this package has not seen a single release yet, we should take the opportunity and require 5.0 from the get go.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// swift-tools-version:4.2
+// swift-tools-version:5.0
```
Comment (weissi) [inline Package.swift:1]: CC @tomerd / @ktoso I'd personally say 5.3. No new software should be written for 5.2 or below

Comment (tomerd) [reply Package.swift:1]: swift-metrics supports 4.2, hence this does too, but I am okay with bumping it here

Comment (weissi) [reply Package.swift:1]: @tomerd / @fabianfett this officially however is major breaking. But if we decide that we ignore that fact, then we should just straight to a very new version ;)

Comment (fabianfett) [reply Package.swift:1]: @weissi You can not major break, if there has never been a release...

Comment (fabianfett) [reply Package.swift:1]: If we move directly to `5.2` or `5.3` packages like AHC won't be able to use this package (as a test dependency for example. see #11). Is this a tradeoff you are willing to make?

Comment (weissi) [reply Package.swift:1]: ah that's bad, then 5.1 I guess :P

Comment (fabianfett) [reply Package.swift:1]: Okay, since AHC requires 5.0, we should target 5.0.

**Review by weissi — APPROVED** [review]

**Pull request merged**